pkg/controller/nstemplatetier/nstemplatetier_controller.go

```diff
@@ -114,72 +115,69 @@ func (r *ReconcileNSTemplateTier) Reconcile(request reconcile.Request) (reconcil
 	}
 
 	if activeTemplateUpdateRequests < MaxPoolSize {
-		members := r.retrieveMemberClusters()
-		for _, member := range members {
-			// create a TemplateUpdateRequest if active count < max pol size
-			// find a MasterUserRecord which is not already up-to-date
-			// and for which there is no TemplateUpdateRequest yet
-			// fetch by subsets of "MaxPoolSize + 1" size until a MasterUserRecord candidate is found
-			murs := toolchainv1alpha1.MasterUserRecordList{}
-			matchingLabels, err := newLabelSelector(instance, *member)
-			if err != nil {
-				logger.Error(err, "unable to get MasterUserRecords to update")
-				return reconcile.Result{}, err
-			}
-			err = r.client.List(context.Background(), &murs,
-				client.InNamespace(instance.Namespace),
-				client.Limit(MaxPoolSize+1),
-				matchingLabels,
-			)
-			if err != nil {
-				logger.Error(err, "unable to get MasterUserRecords to update")
+		// create a TemplateUpdateRequest if active count < max pol size
+		// find a MasterUserRecord which is not already up-to-date
+		// and for which there is no TemplateUpdateRequest yet
+		// fetch by subsets of "MaxPoolSize + 1" size until a MasterUserRecord candidate is found
+		murs := toolchainv1alpha1.MasterUserRecordList{}
+		matchingLabels, err := newLabelSelector(instance)
+		if err != nil {
+			logger.Error(err, "unable to get MasterUserRecords to update")
```

Comment:
Use `wrapErrorWithStatusUpdate()` instead.

Reply:
I'm not sure what you have in mind with regards to the NSTemplateTier status when such an error occurred. Could you provide me with an example?